src/Http/Client/Auth/Digest.php

```diff
@@ -26,13 +27,61 @@
  */
 class Digest
 {
+    /**
+     * Algorithms
+     */
+    public const MD5 = 'MD5';
+    public const SHA_256 = 'SHA-256';
+    public const SHA_512_256 = 'SHA-512-256';
+    public const MD5_SESS = 'MD5-sess';
+    public const SHA_256_SESS = 'SHA-256-sess';
+    public const SHA_512_256_SESS = 'SHA-512-256-sess';
+
+    /**
+     * QOP
+     */
+    public const QOP_AUTH = 'auth';
+    public const QOP_AUTH_INT = 'auth-int';
+
+    /**
+     * Algorithms <-> Hash type
+     */
+    public const ALGORITHMS = [
+        self::MD5 => 'md5',
+        self::SHA_256 => 'sha256',
+        self::SHA_512_256 => 'sha512/256',
+        self::MD5_SESS => 'md5',
+        self::SHA_256_SESS => 'sha256',
+        self::SHA_512_256_SESS => 'sha512/256',
+    ];
     /**
      * Instance of Cake\Http\Client
      *
      * @var \Cake\Http\Client
      */
     protected $_client;
 
+    /**
+     * Algorithm
+     *
+     * @var string
+     */
+    protected $_algorithm;
```

Comment:
We don't user underscore prefix for protected properties/methods anymore.

Reply:
Yeah...it is a shame :(..Removed _ from properties and methods